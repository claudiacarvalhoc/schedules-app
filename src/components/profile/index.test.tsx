import React from 'react';
import Profile, { ProfileProps } from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const getShallowWrapper = (props: ProfileProps) => {
    return shallow(
        <Profile {...props} />
    );
  };

describe('<Profile />', () => {
    it('render correctly with color orange', () => {
        const props: ProfileProps = {
            letter: 'T',
            color: 'orange'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
        expect(component.find('.orange').exists()).toBeTruthy();
    });
    it('render correctly with color purple', () => {
        const props: ProfileProps = {
            letter: 'T',
            color: 'purple'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
        expect(component.find('.purple').exists()).toBeTruthy();
    });
});
