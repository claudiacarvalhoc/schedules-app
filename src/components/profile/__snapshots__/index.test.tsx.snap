// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> render correctly with color orange 1`] = `
<WithStyles(ForwardRef(Avatar))
  className="orange"
>
  T
</WithStyles(ForwardRef(Avatar))>
`;

exports[`<Profile /> render correctly with color purple 1`] = `
<WithStyles(ForwardRef(Avatar))
  className="purple"
>
  T
</WithStyles(ForwardRef(Avatar))>
`;
